config/config.go

```diff
@@ -664,6 +672,34 @@ func (c *KubernetesSDConfig) UnmarshalYAML(unmarshal func(interface{}) error) er
 	return checkOverflow(c.XXX, "kubernetes_sd_config")
 }
 
+// EC2SDConfig is the configuration for EC2 based service discovery.
+type EC2SDConfig struct {
+	Region          string   `yaml:"region"`
+	AccessKey       string   `yaml:"access_key,omitempty"`
+	SecretKey       string   `yaml:"secret_key,omitempty"`
+	RefreshInterval Duration `yaml:"refresh_interval,omitempty"`
+	Port            int      `yaml:"port"`
+	// Catches all undefined fields and must be empty after parsing.
+	XXX map[string]interface{} `yaml:",inline"`
+}
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface.
+func (c *EC2SDConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	*c = DefaultEC2SDConfig
+	type plain EC2SDConfig
+	err := unmarshal((*plain)(c))
+	if err != nil {
+		return err
+	}
+	if c.Region == "" {
+		return fmt.Errorf("EC2 SD configuration requires a region")
+	}
+	if c.AccessKey == "" || c.SecretKey == "" {
+		return fmt.Errorf("EC2 SD configuration requires both access and secret keys")
```

Comment:
Access and secret keys aren't required if you'd getting credentials from the environment.